src/components/KPop/KPop.vue

```diff
@@ -1,431 +1,306 @@
 <template>
   <component
     :is="tag"
-    :id="targetId"
-    ref="root"
-    :aria-controls="$slots.default ? popoverId : undefined"
-    :aria-expanded="$slots.default ? (!!isOpen || undefined) : undefined"
-    :role="$slots.default ? 'button' : null"
-    @keyup.enter.stop.prevent="showPopper"
-    @keyup.esc="hidePopper"
+    ref="kPopoverElement"
+    class="k-popover"
   >
-    <slot>
-      <KButton
-        :aria-controls="popoverId || undefined"
-        :aria-expanded="!!isOpen || undefined"
-        data-testid="popover-button"
-      >
-        {{ buttonText }}
-      </KButton>
-    </slot>
+    <div
+      ref="triggerWrapperElement"
+      class="popover-trigger-wrapper"
+    >
+      <slot>
+        <KButton
+          :aria-controls="popoverId || undefined"
+          data-testid="popover-button"
+        >
+          {{ buttonText }}
+        </KButton>
+      </slot>
+    </div>
+
     <Transition name="kongponents-fade-transition">
       <div
-        v-show="isOpen"
+        v-show="isVisible"
         :id="popoverId"
-        ref="popper"
+        ref="popoverElement"
         :aria-labelledby="$slots.title || title ? titleId : undefined"
-        class="k-popover"
-        :class="popoverClassObj"
+        class="popover"
+        :class="popoverClassesObj"
         role="dialog"
         :style="popoverStyle"
+        :x-placement="calculatedPlacement"
       >
-        <!-- click on close button is handled by handleClick method -->
-        <button
-          v-if="!hideCloseIcon"
-          ref="popoverCloseButton"
-          class="popover-close-button"
-          :tabindex="isOpen ? 0 : -1"
-          type="button"
-        >
-          <CloseIcon
-            class="popover-close-icon"
-            :size="KUI_ICON_SIZE_30"
-          />
-        </button>
-        <div
-          v-if="$slots.title || title"
-          class="popover-header"
-        >
+        <div class="popover-container">
+          <button
+            v-if="!hideCloseIcon"
+            ref="popoverCloseButton"
+            class="popover-close-button"
+            :tabindex="isVisible ? 0 : -1"
+            type="button"
+            @click="hidePopover"
+          >
+            <CloseIcon
+              class="popover-close-icon"
```

Comment:
question: should this icon have a `title` or `aria-label`, or the `decorative` prop?

Reply:
Fixed